base/array.jl

```diff
@@ -656,6 +656,20 @@ function collect_to_with_first!(dest, v1, itr, st)
     return grow_to!(dest, itr, st)
 end
 
+function setindex_widen!(dest::Array{T}, el, i) where T
+    @_inline_meta
+    if el isa T || typeof(el) === T
+        @inbounds dest[i] = el::T
+        dest
```

Comment:
I think we typically recommend `return` for multi line functions with branches etc (for readability reasons).

Reply:
Ok